src/main/java/com/gooddata/warehouse/WarehouseS3Credentials.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2007-2017, GoodData(R) Corporation. All rights reserved.
+ */
+package com.gooddata.warehouse;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.gooddata.util.ISODateTimeDeserializer;
+import com.gooddata.util.ISODateTimeSerializer;
+import org.joda.time.DateTime;
+import org.springframework.web.util.UriTemplate;
+
+import java.util.Map;
+
+import static com.gooddata.util.Validate.notNullState;
+
+/**
+ * Single warehouse S3 credentials record (identified uniquely by warehouse ID, access key and region)
+ */
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonTypeName("s3Credentials")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class WarehouseS3Credentials {
+
+    public static final String URI = WarehouseS3CredentialsList.URI + "/{region}/{accessKey}";
+    public static final UriTemplate TEMPLATE = new UriTemplate(URI);
+    private static final String SELF_LINK = "self";
+
+    private final DateTime modifiedAt;
+
+    private final String userLogin;
+
+    private final String secretKey;
+
+    private final String accessKey;
+
+    private final String region;
+
+    private Map<String, String> links;
+
+    /**
+     * Used to add new S3 credentials
+     *
+     * @param accessKey S3 access key
+     * @param region    S3 region
+     * @param secretKey S3 secret key
+     */
+    public WarehouseS3Credentials(final String accessKey,
+                         final String region,
+                         final String secretKey) {
+        this(accessKey, region, null, secretKey, null, null);
+    }
+
+    /**
+     * Used to list saved S3 credentials
+     *
+     * @param accessKey  S3 access key
+     * @param region     S3 region
+     * @param userLogin  user's login
+     * @param modifiedAt last modification date
+     */
+    public WarehouseS3Credentials(final String accessKey,
```

Comment:
For now I only added a statement explicitly stating that it should not be used by users. I don't think a factory method would change anything, it would still have to be public and might tempt end users to use it.